daemon/cmd/daemon.go

```diff
@@ -392,8 +392,14 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		IPv4: option.Config.EnableIPv4,
 		IPv6: option.Config.EnableIPv6,
 
-		MaxSockRevNatMapEntries: option.Config.SockRevNatEntries,
-		MaxEntries:              option.Config.LBMapEntries,
+		MaxSockRevNatMapEntries:     option.Config.SockRevNatEntries,
+		MaxEntries:                  option.Config.LBMapEntries,
```

Comment:
I think at this point `option.Config.LBMapEntries` is just a configuration detail and should not be passed down to `lbmap.Init()`.

What I would suggest is to move the logic you added in `lbmap.Init()` (https://github.com/koncha99/cilium/blob/aa62b2439e8bdfa72a8a16e080d5621c2e822bb8/pkg/maps/lbmap/lbmap.go#L694-L702) here, get rid of `MaxEntries` from `lbmap.InitParams{}` and make the agent responsible for setting the correct values of the individual  `ServiceMapMaxEntries`, `ServiceBackEndMapMaxEntries`, .. fields.

Moreover we should probably make `--bpf-lb-map-max` mutually exclusive with all the other `--bpf-lb-service-map-max`, `--bpf-lb-service-backend-map-max`, .. options (in order to avoid ambiguous configurations)

Reply:
> Moreover we should probably make --bpf-lb-map-max mutually exclusive with all the other

An alternative solution would be to default all new flags to 0, and if any of it is not set, then set it to `--bpf-lb-map-max`.

Reply:
Thanks, @jibi and @brb for reviewing a PR. Since, as @brb suggested, I marked new options as hidden and keep `--bpf-lb-map-max` as the main option, I made new options to override the value set by `--bpf-lb-map-max` and described their behavior on usage.